src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -748,8 +748,20 @@ def __call__(self):
         cmsswStep = self.step.getTypeHelper()
         for om in cmsswStep.listOutputModules():
             mod = cmsswStep.getOutputModule(om)
+            modName = mod.getInternalName()
+
+            if funcName == 'merge':
+                # Do not use both Merged output label unless useErrorDataset is False
+                # Do not use both MergedError output label unless useErrorDataset is True 
+                useErrorDataset = getattr(self.jobBag, "useErrorDataset", False)
+
+                if useErrorDataset and modName != 'MergedError':
+                    continue
+                if not useErrorDataset and modName == 'MergedError':
+                    continue
+
             makeOutputTweak(mod, self.job, self.tweak)
-        self.applyPsetTweak(self.tweak, cleanupTweak=True)
+        self.applyPsetTweak(self.tweak, allowFailedTweaks=True, cleanupTweak=True)
```

Comment:
A short comment above this line explaining why we allow failed tweaks to be skipped here and why we clean it up would be useful as well.